### PR TITLE
Add reusability with precision + js.Date/java.util.Date/UUID

### PR DESCRIPTION
**Pull request by matthughes**
Float/Double withPrecision plus js.Date/java.util.Date/UUID cause I use those types all the time.  Up to you though.

### CODE REVIEW

File: extra/src/main/scala/japgolly/scalajs/react/extra/Reusability.scala [hunk 1]
```diff
@@ -77,16 +80,23 @@ object Reusability {
...
+  @inline implicit def reusableChar       : Reusability[Char   ] = by_==
+  @inline implicit def reusableShort      : Reusability[Short  ] = by_==
+  @inline implicit def reusableInt        : Reusability[Int    ] = by_==
+  @inline implicit def reusableLong       : Reusability[Long   ] = by_==
+  @inline implicit def reusableString     : Reusability[String ] = by_==
+  @inline implicit def reusableJavaDate   : Reusability[Date   ] = by_==
+  @inline implicit def reusableJsDate     : Reusability[JsDate ] = by_==
+  @inline implicit def reusableUUID       : Reusability[UUID   ] = by_==
+
+  implicit def doubleWithPrecision(precision: Double): Reusability[Double] = fn((x, y) =>
```
Comment (japgolly) [inline extra/src/main/scala/japgolly/scalajs/react/extra/Reusability.scala]: Can't be `implicit` here - it needs an arg.
Can we rename? How about just `def double(tolerance: Double): Reusability[Double]`
And same with float

Comment (japgolly) [inline extra/src/main/scala/japgolly/scalajs/react/extra/Reusability.scala]: Oh and could you just move it up above the `// Instances` line?
By `// Instances` I meant "implicit, default instances"

**Comment by japgolly** [comment]
Otherwise LGTM

**Comment by japgolly** [comment]
Oh, are you sure that univ equality holds for `Date` and `js.Date`? Could you chuck in a quick unit test?

**Comment by matthughes** [comment]
Updated.  Bah, I should have known js.Date wouldn't doesn't equality.  Made other requested changes.

**Comment by japgolly** [comment]
Oh lucky. That turned out well! Thanks

**Pull request merged**